### ISSUE TITLE
[Request] include reason for potential param in the json output

### ISSUE THREAD

**Issue by bugbaba** [Sh1Yo/x8#11]
Привет :) 

Currently, the tooling only mentions the reason a potential parameter is detected in the stdout only. It would be really helpful in cases where a user has multiple output files from the past to see which parameter was detected to reflecting back vs changing in response code, etc.

![image](https://user-images.githubusercontent.com/29960776/135412973-b2eb0b5c-4fef-4943-a182-d35836c00d48.png)

Current output format
```json
{"method":"GET", "url":"http://ffuf.me/cd/param/data", "parameters":["debug"]}
```

Desired output format
```json
{
   "method": "GET",
    "url": "http://ffuf.me/cd/param/data",
    "parameters" : [
        {
            "param": "debug",
            "reason": "Different response code: 400 -> 200"
        }
    ]
}
```

So instead of just sending parameter names to `futures_data`, also send the message that was sent to `writeln`
https://github.com/Sh1Yo/x8/blob/main/src/logic.rs#L216

--
Regards,
@bugbaba

**Comment by Sh1Yo** [comment]
Привет! It looks like a good feature. I will try to find some free time and implement it sometime.

**Comment by bugbaba** [comment]
Looks perfect!

**Comment by Sh1Yo** [comment]
I've published a release v3.1.0 with this improvement.
